### ISSUE TITLE
Check for translation in fallbackLang

### ISSUE THREAD

**Issue by VigoKrumins** [Philipp15b/php-i18n#28]
It doesn't have to check for a string in fallbackLang if there is not translation for string?

E.g.

fallbackLang is "English" but current site language is "Russian". And there is not "greetings" in Russian. Wouldn't it return translation in English instead of returning "Fatal error: Undefined class constant 'greetings'"?

English language JSON:
```
{
  "greetings": "Hello world!",
  "logout": "Log out"
}
```

Russian language JSON:
```
{
  "logout": "выйти"
}
```

**Comment by Philipp15b** [comment]
No, php-i18n always only works with a single language file at a time. That means it doesn't merge two language files and fallback languages are only used if the language file for your other languages can't be found at all.

This could be implemented, but I'm not sure whether this is a good feature. Translations should be complete and not have random strings of other languages in between. If enough people want this feature though, I'd implement it.

**Comment by bocall** [comment]
The implementation of jQuery.i18n defaults to messages.properties for any strings that do not have a translation in a particular language.  I agree that a translation should be complete but if you are looking for assistance for translations it would be a handy feature to have it at least default to something.

**Comment by josephzidell** [comment]
Oh crap. We were about to start integrating this on an enterprise-level app (1500+ views), but this will be an unfortunate deal breaker. Any way this could get done?

We have a large development team who keeps on adding new strings, and the outsourced translation team is slightly behind.

**Comment by denizdogan** [comment]
This might be a deal-breaker for me too. I don't have a *huge* app, but there's lots and lots of translation that'll be needed.

@josephzidell Have you come to some conclusion about how to deal with this? I'm thinking as a last resort, I might implement something which automatically merges the translations files outside of this library.

**Comment by Philipp15b** [comment]
Since this does seem to be a more widely requested feature, I'll start with the implementation of an option to merge fallback language strings into other languages.

**Comment by Philipp15b** [comment]
Fixed by 738bbe1d6580eaaabdc5006edc72a6767dc1115f.

**Comment by josephzidell** [comment]
We start implementation today. Thanks for fixing this. 👍
